open Android's share sheet from "More"

"More" in the share sheet prepares the selected items like an app tap does, then starts
Intent.createChooser() with the prepared share intent instead of the OPLUS "All apps" panel.
Helper class: com.oplus.gallery.aosp.AospShare (source in src/gallery).

diff --git b/classes2/com/oplus/gallery/aosp/AospShare.smali b/classes2/com/oplus/gallery/aosp/AospShare.smali
new file mode 100644
index 0000000..23c0aaf
--- /dev/null
+++ b/classes2/com/oplus/gallery/aosp/AospShare.smali
@@ -0,0 +1,146 @@
+.class public final Lcom/oplus/gallery/aosp/AospShare;
+.super Ljava/lang/Object;
+.source "AospShare.java"
+
+# interfaces
+.implements Lkotlin/jvm/functions/Function1;
+
+
+# annotations
+.annotation system Ldalvik/annotation/Signature;
+    value = {
+        "Ljava/lang/Object;",
+        "Lkotlin/jvm/functions/Function1<",
+        "Ljava/lang/Object;",
+        "Ljava/lang/Object;",
+        ">;"
+    }
+.end annotation
+
+
+# static fields
+.field private static final TAG:Ljava/lang/String; = "AospShare"
+
+
+# instance fields
+.field private final mActivity:Landroid/app/Activity;
+
+.field private final mViewModel:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;
+
+
+# direct methods
+.method private constructor <init>(Landroid/app/Activity;Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;)V
+    .locals 0
+
+    .line 20
+    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    .line 21
+    iput-object p1, p0, Lcom/oplus/gallery/aosp/AospShare;->mActivity:Landroid/app/Activity;
+
+    .line 22
+    iput-object p2, p0, Lcom/oplus/gallery/aosp/AospShare;->mViewModel:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;
+
+    .line 23
+    return-void
+.end method
+
+.method public static more(Landroid/app/Activity;Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;)V
+    .locals 2
+
+    .line 27
+    if-eqz p0, :cond_10
+
+    if-nez p1, :cond_5
+
+    goto :goto_10
+
+    .line 30
+    :cond_5
+    sget-object v0, Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel$DownloadTriggerType;->TRIGGER_RESOLVE_INFO:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel$DownloadTriggerType;
+
+    new-instance v1, Lcom/oplus/gallery/aosp/AospShare;
+
+    invoke-direct {v1, p0, p1}, Lcom/oplus/gallery/aosp/AospShare;-><init>(Landroid/app/Activity;Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;)V
+
+    invoke-virtual {p1, p0, v0, v1}, Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;->D0(Landroid/app/Activity;Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel$DownloadTriggerType;Lkotlin/jvm/functions/Function1;)V
+
+    .line 32
+    return-void
+
+    .line 28
+    :cond_10
+    :goto_10
+    return-void
+.end method
+
+
+# virtual methods
+.method public invoke(Ljava/lang/Object;)Ljava/lang/Object;
+    .locals 2
+
+    .line 36
+    iget-object v0, p0, Lcom/oplus/gallery/aosp/AospShare;->mViewModel:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;
+
+    iget-object v0, v0, Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;->w0:Landroid/content/Intent;
+
+    .line 37
+    if-eqz p1, :cond_2a
+
+    if-nez v0, :cond_9
+
+    goto :goto_2a
+
+    .line 41
+    :cond_9
+    new-instance p1, Landroid/content/Intent;
+
+    invoke-direct {p1, v0}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
+
+    .line 42
+    const/4 v0, 0x0
+
+    invoke-virtual {p1, v0}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
+
+    .line 43
+    invoke-virtual {p1, v0}, Landroid/content/Intent;->setPackage(Ljava/lang/String;)Landroid/content/Intent;
+
+    .line 45
+    :try_start_15
+    iget-object v1, p0, Lcom/oplus/gallery/aosp/AospShare;->mActivity:Landroid/app/Activity;
+
+    invoke-static {p1, v0}, Landroid/content/Intent;->createChooser(Landroid/content/Intent;Ljava/lang/CharSequence;)Landroid/content/Intent;
+
+    move-result-object p1
+
+    invoke-virtual {v1, p1}, Landroid/app/Activity;->startActivity(Landroid/content/Intent;)V
+    :try_end_1e
+    .catch Ljava/lang/RuntimeException; {:try_start_15 .. :try_end_1e} :catch_1f
+
+    .line 48
+    goto :goto_27
+
+    .line 46
+    :catch_1f
+    move-exception p1
+
+    .line 47
+    const-string v0, "AospShare"
+
+    const-string v1, "cannot start the chooser"
+
+    invoke-static {v0, v1, p1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+
+    .line 49
+    :goto_27
+    sget-object p1, Lkotlin/Unit;->INSTANCE:Lkotlin/Unit;
+
+    return-object p1
+
+    .line 38
+    :cond_2a
+    :goto_2a
+    sget-object p1, Lkotlin/Unit;->INSTANCE:Lkotlin/Unit;
+
+    return-object p1
+.end method
diff --git a/classes2/com/oplus/gallery/sharepage/view/ShareViewBinding.smali b/classes2/com/oplus/gallery/sharepage/view/ShareViewBinding.smali
index 00f4c59..61a0a24 100644
--- a/classes2/com/oplus/gallery/sharepage/view/ShareViewBinding.smali
+++ b/classes2/com/oplus/gallery/sharepage/view/ShareViewBinding.smali
@@ -3075,7 +3075,7 @@
 
     .line 2
     .line 3
-    if-nez v0, :cond_90
+    if-nez v0, :cond_96
 
     .line 4
     .line 5
@@ -3095,7 +3095,7 @@
 
     .line 12
     .line 13
-    goto/16 :goto_90
+    goto/16 :goto_96
 
     .line 14
     .line 15
@@ -3142,7 +3142,7 @@
 
     .line 32
     .line 33
-    goto/16 :goto_90
+    goto/16 :goto_96
 
     .line 34
     .line 35
@@ -3163,7 +3163,7 @@
 
     .line 42
     .line 43
-    if-eqz v1, :cond_90
+    if-eqz v1, :cond_96
 
     .line 44
     .line 45
@@ -3175,7 +3175,7 @@
 
     .line 48
     .line 49
-    goto :goto_90
+    goto :goto_96
 
     .line 50
     :cond_31
@@ -3190,7 +3190,7 @@
 
     .line 53
     .line 54
-    if-eqz v1, :cond_90
+    if-eqz v1, :cond_96
 
     .line 55
     .line 56
@@ -3242,7 +3242,7 @@
 
     .line 77
     .line 78
-    goto :goto_90
+    goto :goto_96
 
     .line 79
     :goto_4e
@@ -3287,7 +3287,13 @@
     move-result v0
 
     .line 99
-    if-eqz v0, :cond_7f
+    if-eqz v0, :cond_85
+
+    iget-object p0, p0, Lcom/oplus/gallery/sharepage/view/ShareViewBinding;->B0:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;
+
+    invoke-static {v7, p0}, Lcom/oplus/gallery/aosp/AospShare;->more(Landroid/app/Activity;Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;)V
+
+    return-void
 
     .line 100
     .line 101
@@ -3342,12 +3348,12 @@
     return-void
 
     .line 128
-    :cond_7f
+    :cond_85
     iget-object v0, p0, Lcom/oplus/gallery/sharepage/view/ShareViewBinding;->B0:Lcom/oplus/gallery/sharepage/viewmodel/ShareInnerViewModel;
 
     .line 129
     .line 130
-    if-eqz v0, :cond_90
+    if-eqz v0, :cond_96
 
     .line 131
     .line 132
@@ -3378,8 +3384,8 @@
     .line 143
     .line 144
     .line 145
-    :cond_90
-    :goto_90
+    :cond_96
+    :goto_96
     return-void
     .line 146
     .line 147
